raft.go

```diff
@@ -452,15 +447,15 @@ func (r *Raft) VerifyLeader() Future {
 
 // AddPeer (deprecated) is used to add a new peer into the cluster. This must be
 // run on the leader or it will fail. Use AddVoter/AddNonvoter instead.
-func (r *Raft) AddPeer(peer string) Future {
-	return r.AddVoter(ServerID(peer), ServerAddress(peer), 0, 0)
+func (r *Raft) AddPeer(peer ServerAddress) Future {
+	return r.AddVoter(ServerID(peer), peer, 0, 0)
 }
 
 // RemovePeer (deprecated) is used to remove a peer from the cluster. If the
 // current leader is being removed, it will cause a new election
 // to occur. This must be run on the leader or it will fail.
 // Use RemoveServer instead.
-func (r *Raft) RemovePeer(peer string) Future {
+func (r *Raft) RemovePeer(peer ServerAddress) Future {
```

Comment:
I originally wanted to keep this as `string` but this one was confusing (it's an address that we assume is the same as the ID). The trivial work to fix this is probably worth the clarity that the type provides.